plugins/hls-tactics-plugin/src/Ide/Plugin/Tactic/GHC.hs

```diff
@@ -112,6 +122,102 @@ algebraicTyCon (splitTyConApp_maybe -> Just (tycon, _))
   | otherwise = Just tycon
 algebraicTyCon _ = Nothing
 
+
+------------------------------------------------------------------------------
+-- | We can't compare 'RdrName' for equality directly. Instead, compare them by
+-- their 'OccName's.
+eqRdrName :: RdrName -> RdrName -> Bool
+eqRdrName = (==) `on` occNameString . occName
```

Comment:
Each `RdrName` is `Located`.  The locations can be used as an index between the identifier location in the Parsed AST and the Renamed one, to get the exact identity, via the `unique` associated with a `Name`.

We used to construct this name map in HIE, it may still be in HLS.  If not, [here](https://github.com/alanz/HaRe/blob/98f390b6e9d48537429863ca890aa853afcd7c79/src/Language/Haskell/Refact/Utils/MonadFunctions.hs#L487) is it's source.

Reply:
Sorry, I'm not sure what you mean by this. The problem I was running into is that the RdrNames that come from `VarPat`s have different locations than the RdrNames from `HsVar`s.

Reply:
For each `Located RdrName`, if you look it up in the nameMap you will get the corresponding `Name`. If they are the same, as identified by the renaming phase, they will both have the same [nameUnique](https://downloads.haskell.org/~ghc/8.8.4/docs/html/libraries/ghc-8.8.4/Name.html#v:nameUnique) value.

Reply:
Upon further thought, I'm not sure this will work for me. This is all a pass over parsed trees, before the renamer has had a chance to run. Since we're simplifying tactics' code, which doesn't ever shadow variables, I don't think this is a concern.